app_feup/lib/view/Widgets/bug_report_form.dart

```diff
@@ -169,83 +186,113 @@ class BugReportFormState extends State<BugReportForm> {
         style: TextStyle(color: Colors.white, fontSize: 20.0),
       ),
-      style: ElevatedButton.styleFrom(
-        padding: EdgeInsets.symmetric(vertical: 10.0),
-        primary: Theme.of(context).primaryColor,
-      ),
     ));
   }
 
-  void submitBugReport() {
+  void submitBugReport() async {
     setState(() {
       _isButtonTapped = true;
     });
 
+    final String title = titleController.text;
     final String bugLabel = bugDescriptions[_selectedBug] == null
         ? 'Unidentified bug'
         : bugDescriptions[_selectedBug].item2;
-    final Map data = {
-      'headline6': titleController.text,
-      'body': descriptionController.text,
-      'labels': [_issueLabel, bugLabel]
-    };
-
-    final url = _postUrl + '?access_token=' + ghToken;
-    http
-        .post(url.toUri(),
-            headers: {'Content-Type': 'application/json'},
-            body: json.encode(data))
-        .then((http.Response response) {
-      final int statusCode = response.statusCode;
-
-      String msg;
-      if (statusCode < 200 || statusCode > 400) {
-        Logger().e('Error ' + statusCode.toString() + ' while posting bug');
-        msg = 'Ocorreu um erro no envio';
-      } else {
-        Logger().i('Successfully submitted bug report.');
-        msg = 'Enviado com sucesso';
-
-        clearForm();
-
-        Navigator.pop(context);
-        setState(() {
-          _isButtonTapped = false;
-        });
-      }
-      
-
-      FocusScope.of(context).requestFocus( FocusNode());
-      displayBugToast(msg);
-      setState(() {
-        _isButtonTapped = false;
-      });
-    }).catchError((error) {
-      Logger().e(error);
-      FocusScope.of(context).requestFocus( FocusNode());
-
-      final String msg =
-          (error is SocketException) ? 'Falha de rede' : 'Ocorreu um erro';
-      displayBugToast(msg);
-      setState(() {
-        _isButtonTapped = false;
+
+    String toastMsg;
+    try {
+      final String sentryDescription = emailController.text == ''
+          ? descriptionController.text
+          : descriptionController.text + '\nContact: ' + emailController.text;
+      final sentryId = await Sentry.captureMessage(
+          bugLabel + ': ' + title + '\n' + sentryDescription);
+
+      final String gitHubDescription =
+          descriptionController.text + '\n' + _sentryLink + sentryId.toString();
+      final Map gitHubData = {
+        'title': titleController.text,
+        'body': gitHubDescription,
+        'labels': [_issueLabel, bugLabel],
+      };
+      http
+          .post(Uri.parse(_gitHubPostUrl),
+              headers: {
+                'Content-Type': 'application/json',
+                'Authorization': 'token $ghToken'
+              },
+              body: json.encode(gitHubData))
+          .then((http.Response response) {
+        final int statusCode = response.statusCode;
+        if (statusCode < 200 || statusCode > 400) {
+          throw Exception('Network error');
+        }
```

Comment:
Can we extract this code into two methods? It would make it much clear to read. Both could return promised and we would wait for both of them at the same time (see the [Future.wait() method](https://api.dart.dev/stable/2.12.2/dart-async/Future/wait.html))

Reply:
I think we need the sentry one to finish first, so that the id is returned to be sent to git?

Reply:
Ah right, well both futures can be chained with a flatmap call nonetheless

Reply:
Do you mean chaining with `then()`? I've searched for `flatmap` in Dart, but only got Scala results :) Nevertheless, I think it's quite clean now, check it out

Reply:
I thought dart also had some api for it but I may be wrong, the way it is fine nonetheless